packages/backend-function/src/lambda-shims/invoke_ssm_shim.ts

```
@@ -9,5 +9,7 @@ await internalAmplifyFunctionResolveSsmParams();
 const SSM_PARAMETER_REFRESH_MS = 1000 * 60;
 
 setInterval(() => {
-  void internalAmplifyFunctionResolveSsmParams();
+  // Catch errors and do nothing in the case we are retrieving secrets when the Lambda starts shutting down
+  // eslint-disable-next-line promise/prefer-await-to-then
+  void internalAmplifyFunctionResolveSsmParams().catch(() => {});
```

Two things.

1. Can we try-catch inside `internalAmplifyFunctionResolveSsmParams` ?
2. Should we at least `console.debug` this ?

---

Re 1. Peharps we don't because we want lambda to crash at cold start.

But maybe there's some way to have normal try catch here ?

---

For 1, yeah we would want it to crash at cold start, just not for subsequent SSM calls. I'll look into having this as a normal try catch so we can remove the eslint comment.

For 2, any logging would cause an `EPIPE` error in the scenario for this error.

---

> For 2, any logging would cause an EPIPE error in the scenario for this error.

Unless... we try-catch logger as well. 

The thing I'm worried about is that empty catch block will eat legit errors that are not due to shutdown.

---

Seems like it's not possible, tried doing
```ts
setInterval(() => {
  try {
    void internalAmplifyFunctionResolveSsmParams();
  } catch (error) {
    // Catch errors and do nothing in the case we are retrieving secrets when the Lambda starts shutting down
  }
}, SSM_PARAMETER_REFRESH_MS);
```
and Lambda doesn't seem to handle top level try catch gracefully and gives us the original error.

---

you would need this

```ts
setInterval(async () => {
  try {
    await internalAmplifyFunctionResolveSsmParams();
  } catch (error) {
    // Catch errors and do nothing in the case we are retrieving secrets when the Lambda starts shutting down
  }
}, SSM_PARAMETER_REFRESH_MS);
```

---

```
setInterval(void (async () => {
  try {
    await internalAmplifyFunctionResolveSsmParams();
  } catch (error) {
    // Catch errors and do nothing in the case we are retrieving secrets when the Lambda starts shutting down
  }
}), SSM_PARAMETER_REFRESH_MS);
```

---

...I may still be in weekend mode, of course it has to be like this.